src/lib.rs

```diff
@@ -2096,6 +2096,38 @@ impl Build {
             tool
         };
 
+        // New "standalone" C/C++ cross-compiler executables from recent Android NDK
+        // are just shell scripts that call main clang binary (from Android NDK) with
+        // proper `--target` argument.
+        //
+        // For example, armv7a-linux-androideabi16-clang passes
+        // `--target=armv7a-linux-androideabi16` to clang.
+        //
+        // As the shell script calls the main clang binary, the command line limit length
+        // on Windows is restricted to around 8k characters instead of around 32k characters.
+        // To remove this limit, we call the main clang binary directly and contruct the
+        // `--target=` ourselves.
+        if host.contains("windows") && android_clang_compiler_uses_target_arg_internally(&tool.path)
+        {
+            let mut f = || -> Option<bool> {
+                let file_name = tool.path.file_name()?.to_str()?.to_owned();
+                let (target, clang) = file_name.split_at(file_name.rfind("-")?);
+
+                tool.path.set_file_name(clang.trim_start_matches("-"));
+                tool.path.set_extension("exe");
+                tool.args.push(format!("--target={}", target).into());
+
+                for version in 16..=23 {
+                    if target.contains(&format!("i686-linux-android{}", version)) {
+                        tool.args.push("-mstackrealign".into());
+                    }
+                }
+
+                Some(true)
+            };
+            f();
```

Comment:
I think the inner closure should be ok here to remove, it's fine to just have one `if let` for the `file_name` binding and avoid the closure business.